### PR TITLE
ColorSliderGroup - refactor

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
ColorSliderGroup - refactor

## Changelog
ColorSliderGroup - refactor

## Additional info
ticket 4107

### CODE REVIEW

**Review by ethanshar — APPROVED** [review]

File: src/components/slider/GradientSlider.tsx [hunk 1]
```diff
@@ -172,6 +157,5 @@ GradientSlider.displayName = 'GradientSlider';
 GradientSlider.types = GradientSliderTypes;
 // @ts-expect-error
 export default asBaseComponent<GradientSliderProps, ComponentStatics<typeof GradientSlider>>(
```
Comment (ethanshar) [inline src/components/slider/GradientSlider.tsx:159]: We can add another simplification 
Replace asBaseComponent HOC with the `useThemeProps` (see usages in View component)

**Review by ethanshar — COMMENTED** [review]
Looks good. Wrote couple of comments, also I noticed a weird bug, I'm not sure if it's a regression, but there's a small flicker when you drag of the sliders in the group

See attached video 
https://github.com/wix/react-native-ui-lib/assets/1780255/5bf210f5-68c5-404f-b9f8-e418caf3c5dd

File: src/components/slider/GradientSlider.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/components/slider/GradientSlider.tsx:159]: I got issues with the statics and the forwardRef so I decided to leave it

File: src/components/slider/GradientSlider.tsx [hunk 2]
```diff
@@ -172,6 +157,5 @@ GradientSlider.displayName = 'GradientSlider';
...
-  // @ts-expect-error
-  forwardRef(asSliderGroupChild(forwardRef(GradientSlider)))
+  forwardRef(forwardRef(GradientSlider))
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/slider/GradientSlider.tsx]: Removed

**Comment by Inbal-Tish** [comment]
> Looks good. Wrote couple of comments, also I noticed a weird bug, I'm not sure if it's a regression, but there's a small flicker when you drag of the sliders in the group
> 
> See attached video https://github.com/wix/react-native-ui-lib/assets/1780255/5bf210f5-68c5-404f-b9f8-e418caf3c5dd

Fixed

**Comment by ethanshar** [comment]
> > Looks good. Wrote couple of comments, also I noticed a weird bug, I'm not sure if it's a regression, but there's a small flicker when you drag of the sliders in the group
> > See attached video https://github.com/wix/react-native-ui-lib/assets/1780255/5bf210f5-68c5-404f-b9f8-e418caf3c5dd
> 
> Fixed

Check if ColorPicker works for you? 
I can't drag the gradient sliders in ColorPicker

File: src/components/slider/GradientSlider.tsx (hunk 1 above)
Comment (ethanshar) [reply src/components/slider/GradientSlider.tsx]: That works for me or at least It didn't throw any errors...
```
export default forwardRef<GradientSliderProps<string | HSLA>, ComponentStatics<typeof GradientSlider>>(GradientSlider);
```

**Comment by Inbal-Tish** [comment]
> > > Looks good. Wrote couple of comments, also I noticed a weird bug, I'm not sure if it's a regression, but there's a small flicker when you drag of the sliders in the group
> > > See attached video https://github.com/wix/react-native-ui-lib/assets/1780255/5bf210f5-68c5-404f-b9f8-e418caf3c5dd
> > 
> > 
> > Fixed
> 
> Check if ColorPicker works for you? I can't drag the gradient sliders in ColorPicker

I can move them... wired

File: src/components/slider/GradientSlider.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/components/slider/GradientSlider.tsx]: Yep, I didn't move the forwardRef... I'll make the change

**Review by ethanshar — APPROVED** [review]

**Pull request merged**